hash.c

```diff
@@ -6323,15 +6355,38 @@ env_update_block_i(VALUE key, VALUE val, VALUE _)
 
 /*
  * call-seq:
- *   ENV.update(hash)                                        -> ENV
- *   ENV.update(hash) { |name, old_value, new_value| block } -> ENV
- *   ENV.merge!(hash)                                        -> ENV
- *   ENV.merge!(hash) { |name, old_value, new_value| block } -> ENV
- *
- * Adds the contents of +hash+ to the environment variables.  If no block is
- * specified entries with duplicate keys are overwritten, otherwise the value
- * of each duplicate name is determined by calling the block with the key, its
- * value from the environment and its value from the hash.
+ *   ENV.update(hash)                                     -> ENV
+ *   ENV.update(hash) { |name, env_val, hash_val| block } -> ENV
+ *   ENV.merge!(hash)                                     -> ENV
+ *   ENV.merge!(hash) { |name, env_val, hash_val| block } -> ENV
+ *
+ * ENV.update is an alias for ENV.merge!.
+ *
+ * Adds to ENV each key/value pair in the given +hash+; returns ENV:
+ *   ENV.replace('foo' => '0', 'bar' => '1')
+ *   ENV.merge!('baz' => '2', 'bat' => '3') # => {"bar"=>"1", "bat"=>"3", "baz"=>"2", "foo"=>"0"}
+ * Deletes the ENV entry for a hash value that is +nil+:
+ *   ENV.merge!('baz' => nil, 'bat' => nil) # => {"bar"=>"1", "foo"=>"0"}
+ * For an already-existing name, if no block given, overwrites the ENV value:
+ *   ENV.merge!('foo' => '4') # => {"bar"=>"1", "foo"=>"4"}
+ * For an already-existing name, if block given,
+ * yields the name, its ENV value, and its hash value;
+ * the  block's return value becomes the new name:
+ *   ENV.merge!('foo' => '5') { |name, env_val, hash_val | env_val + hash_val } # => {"bar"=>"1", "foo"=>"45"}
+ * Raises an exception if a name or value is invalid
+ * (see {Invalid Names and Values}[#class-ENV-label-Invalid-Names+and+Values]);
```

Comment:
Several Errors existed in the file with the linking to the Class 'Introduction' headings.

All the id's are of the form:
```
#class-ENV-label-<heading text>
```
With spaces in \<heading text\> replaced by plus signs.

So the link should be:
```
[#class-ENV-label-Invalid+Names+and+Values
```

Reply:
Oh, wow!  Thanks MSP-Greg.  Correcting now.

Reply:
Thanks for working on docs.  Sorry for the above, as I didn't mention that I was referring to issues with RDoc, etc.  You made the connection...

I noticed it with:

https://msp-greg.github.io/ruby_trunk/Core/ENV.html

then checked

https://docs.ruby-lang.org/en/master/ENV.html

Reply:
I think it's fixed now.